src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientLSBatchOpsImpl.java

```diff
@@ -213,88 +307,126 @@ public List<Object> executeWithResult() throws Exception {
-        for (ObPair<Integer, ObTableSingleOp> operationWithIndex : subOperationWithIndexList) {
-            subOperations.add(operationWithIndex.getRight());
+        long lsId = lsOperation.getKey();
+        Map<Long, ObPair<ObTableParam, List<ObPair<Integer, ObTableSingleOp>>>> tabletOperationsMap = lsOperation.getValue();
+        if (tabletOperationsMap.size() == 0) {
+            logger.warn("the size of tablet operations in ls operation is zero");
+            throw new ObTableUnexpectedException("the size of tablet operations in ls operation is zero");
+        }
+
+        ObTableLSOpRequest tableLsOpRequest = new ObTableLSOpRequest();
+        ObTableLSOperation tableLsOp = new ObTableLSOperation();
+        tableLsOp.setLsId(lsId);
+        tableLsOp.setReturnOneResult(returnOneResult);
+        // fetch the following parameters in first entry for routing
+        long tableId = 0;
+        long originPartId = 0;
+        long operationTimeout = 0;
+        ObTable subObTable = null;
+
+        boolean isFirstEntry = true;
+        // list ( index list for tablet op 1, index list for tablet op 2, ...)
+        List<List<ObPair<Integer, ObTableSingleOp>>> lsOperationWithIndexList = new ArrayList<>();
+        for (final Map.Entry<Long, ObPair<ObTableParam, List<ObPair<Integer, ObTableSingleOp>>>> tabletOperation : tabletOperationsMap.entrySet()) {
+            ObTableParam tableParam = tabletOperation.getValue().getLeft();
+            long tabletId = tableParam.getPartitionId();
+            List<ObPair<Integer, ObTableSingleOp>> tabletOperationWithIndexList = tabletOperation.getValue().getRight();
+            lsOperationWithIndexList.add(tabletOperationWithIndexList);
+            List<ObTableSingleOp> singleOps = new ArrayList<>();
+            for (ObPair<Integer, ObTableSingleOp> operationWithIndex : tabletOperationWithIndexList) {
+                singleOps.add(operationWithIndex.getRight());
+            }
+            ObTableTabletOp tableTabletOp = new ObTableTabletOp();
+            tableTabletOp.setSingleOperations(singleOps);
+            tableTabletOp.setTabletId(tabletId);
+
+            tableLsOp.addTabletOperation(tableTabletOp);
+
+            if (isFirstEntry) {
+                tableId = tableParam.getTableId();
+                originPartId = tableParam.getPartId();
+                operationTimeout = tableParam.getObTable().getObTableOperationTimeout();
+                subObTable = tableParam.getObTable();
+                isFirstEntry = false;
+            }
         }
-        ObTableTabletOp tabletOp = new ObTableTabletOp();
-        tabletOp.setSingleOperations(subOperations);
-        tabletOp.setTabletId(partId);
 
-        ObTableLSOperation lsOperation = new ObTableLSOperation();
-        lsOperation.addTabletOperation(tabletOp);
         // Since we only have one tablet operation
         // We do the LS operation prepare here
-        lsOperation.prepare();
+       tableLsOp.prepare();
```

Comment:
Make sure that `tableLsOp.prepare()` is still working on multi-tabletOP and modify the comment above.

Reply:
yeah, there will be a for-loop in tablet ops to construct column maps and options.